LINK-1170 | Authorization code flow for authentication

jorilindell opened the pull request:
## Description :sparkles:
- Refactor next-auth to use Authorization code flow instead of implicit flow
- Don't store api token to cookie but use session information instead
- Instructions to run local development environment with Tunnistamo, Linked Events API and Linked Registration UI

## Issues :bug:

### Closes :no_good_woman:

[LINK-1170](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1170)

### Related :handshake:

## Testing :alembic:
At the moment [Tunnistamo](https://api.hel.fi/sso) is not configured to have client for Linked Registration UI. At the moment authentication can only be tested locally. Please follow instructions at "Setting up complete development environment locally with docker" section of README.md to test it

[LINK-1170]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1170?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mikkojamG reviewed (commented):
LGTM 👍 Looks like the test coverage doesn't quite reach 80% mark ATM 🤔

jorilindell commented:
> LGTM 👍 Looks like the test coverage doesn't quite reach 80% mark ATM 🤔

Added getPublicRuntimeConfig and getServerRuntimeConfig helper functions to unify way to get environment variables. Added also mockConfig and mockDefaultConfig functions to mock runtime configurations for the unit tests
